packages/eas-cli/src/project/publish.ts

```diff
@@ -127,55 +134,36 @@ export function resolveInputDirectory(customInputDirectory: string): string {
   return distRoot;
 }
 
-export function collectUserDefinedAssets(distRoot: string): RawAsset[] {
-  const assetRoot = path.join(distRoot, 'assets');
-  const assetPointers = Platforms.map(platform => {
-    const assetJsonPath = path.join(distRoot, `${platform}-index.json`);
-    return JsonFile.read(assetJsonPath).bundledAssets;
-  }).flat();
-
-  return [...new Set(assetPointers)].map(pointer => {
-    const [filename, ext] = new String(pointer ?? '').split('_').pop()?.split('.') ?? [];
-    if (!filename) {
-      throw new Error(
-        'There was an error locating all of the bundler defined assets. Please rerun the bundler and then retry publishing.'
-      );
-    }
-    return {
-      type: ext ?? '',
-      contentType: guessContentTypeFromExtension(ext),
-      path: path.join(assetRoot, filename),
-    };
-  });
-}
-
-export function collectBundles(distRoot: string): { [key: string]: RawAsset } {
-  const bundleRoot = path.join(distRoot, 'bundles');
-  const bundlePaths = Object.fromEntries(
-    fs.readdirSync(bundleRoot).map(name => [name.split('-')[0], path.join(bundleRoot, name)])
-  );
-
-  const bundleBuffers: { [key: string]: RawAsset } = {};
-  Platforms.forEach(platform => {
-    bundleBuffers[platform] = {
-      type: 'bundle',
-      contentType: 'application/javascript',
-      path: bundlePaths[platform],
-    };
-  });
-  return bundleBuffers;
+export function loadMetadata(distRoot: string): Metadata {
+  const metadata: Metadata = JsonFile.read(path.join(distRoot, 'metadata.json'));
+  if (metadata.version !== 0) {
+    throw new Error('Only bundles with metadata version 0 are supported');
+  }
+  if (metadata.bundler !== 'metro') {
+    throw new Error('Only bundles created by metro are supported');
+  }
+  return metadata;
 }
 
 export function collectAssets(inputDir: string): CollectedAssets {
   const distRoot = resolveInputDirectory(inputDir);
-  const assets = collectUserDefinedAssets(distRoot);
-  const bundles = collectBundles(distRoot);
-  const assetsFinal: CollectedAssets = {};
+  const metadata = loadMetadata(distRoot);
 
+  const assetsFinal: CollectedAssets = {};
   for (const platform of Platforms) {
     assetsFinal[platform] = {
-      launchAsset: bundles[platform],
-      assets,
+      launchAsset: {
+        type: 'bundle',
+        contentType: 'application/javascript',
+        path: path.resolve(distRoot, metadata.fileMetadata[platform].bundle),
```

Comment:
Looks like this might result in `Cannot read property 'bundle' of undefined`, if the bundle doesn't include all platforms? I wonder if we should check that the key exist?

Reply:
good point, added validation with @hapi/joi